File: find_the_access_codes.java
package com.google.challenges; 
import java.lang.Math.*;
import java.util.Arrays;

public class Answer {   
    public static int answer(int[] l) { 
        // Your code goes here
        int n = l.length;
        int[] left = new int[n];
        int[] right = new int[n];
	Arrays.fill(left, 0);
	Arrays.fill(rightt, 0);
        /* 
        left: # of elements is divisable by l[j]
        right: # of elements l[i] is divisable by
        */
        for (int i = 0; i < l.length - 1; i++) 
            for (int j = i + 1; j < l.length; j++) 
            if (l[j] % l[i] == 0) {
		left[j]++;
		right[i]++;
            }
            
        int res = 0;
        for (int i = 1; i < l.length - 1; i++)
            res += left[i] * right[i];
            
        return res;
    } 
}

File: find_the_access_codes.py
answer(l):
    # your code here
    n = len(l)
    left = [0] * n
    right = [0] * n
    for i in range(n - 1):
        for j in range(i + 1, n):
            if (l[j] % l[i] == 0):
                left[j] += 1
                right[i] += 1
    return sum(left[i] * right[i] for i in range(1, n - 1))
